### PR TITLE
[FIX] website: fix dynamic routing

### PR DESCRIPTION
**Pull request by JKE-be**
This commit of merge fix multiple issues in the dynamic routing.

Fix RequestUID error in 308 redirect
Support MissingError / unexisting ID in 308 redirect
Correctly rewrite url on rendering by url_for for non editor user

Related to opw-2420562

### CODE REVIEW

**Review by rdeodoo — APPROVED** [review]
LGTM, added tests that ensure those 3 bugs are fixed (tests crashes without the fixes)

File: addons/test_website/tests/test_redirect.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            'url_to': '/redirected/country/<model("res.country"):country>',
+        })
+
+    def test_01_redirect_308_model_converter(self):
+        """ Ensure 308 redirect with model converter works fine, including:
+                - Correct & working redirect as public user
+                - Correct & working redirect as logged in user
+                - Correct replace of url_for() URLs in DOM
+        """
+        url = '/test_website/country/andorra-1'
```
Comment (JKE-be) [inline addons/test_website/tests/test_redirect.py:33]: Andorra == id 1 is a little short.
Maybe search the first country e.g. ? or create one ...?

File: addons/test_website/data/test_website_data.xml [hunk 2]
```diff
@@ -95,5 +95,12 @@
             <div>I am a base view</div>
         </template>
 
+        <!-- RECORDS FOR REDIRECT TESTS -->
+        <template id="test_redirect_view">
+            <t t-esc="country.name"/>
+            <t t-if="not request.env.user._is_public()" t-esc="'Logged In'"/>
+            <a t-att-href="url_for('/test_website/country/andorra-1')">I am a link</a>
```
Comment (JKE-be) [inline addons/test_website/data/test_website_data.xml:102]: can't we test classic rendering, with simple href instead ?

Url_for should be called automatically by _post_processing_att

**Review by JKE-be — COMMENTED** [review]
Nice, thanks for test

File: addons/test_website/tests/test_redirect.py (hunk 1 above)
Comment (rdeodoo) [reply addons/test_website/tests/test_redirect.py]: Indeed I was a bit too lazy here

File: addons/test_website/data/test_website_data.xml (hunk 2 above)
Comment (rdeodoo) [reply addons/test_website/data/test_website_data.xml]: I didn't know/forgot `href` weren't send through `url_for` if you were logged in

**Comment by JKE-be** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**